reports/tests.py

```diff
@@ -0,0 +1,504 @@
+                'days': expected_days,
+                'charge': settings.BILLING_RATE * expected_days,
+            })
+
+    def get_instance_charges_future_month(self):
+        """
+        Test that the instance will return empty charges and 0 total for its
+        AppServers if the invoice month is in future.
+        """
+        invoice_month = self._generate_invoice_month(year=2019)
+        appservers_charges, appservers_total = get_instance_charges(OpenEdXInstanceFactory(), invoice_month)
+        self.assertEqual(appservers_charges, [])
+        self.assertEqual(appservers_total, 0)
+
+    @mock.patch('reports.helpers.generate_charge_details')
+    def test_get_instance_charges(self, charges_details_mock):
+        """
+        Test that an instance is going to generate charges for all of its
+        appservers only.
+        """
+        desired_instance = OpenEdXInstanceFactory()
+        make_test_appserver(instance=desired_instance)
+        make_test_appserver(instance=desired_instance)
+
+        another_instance = OpenEdXInstanceFactory()
+        make_test_appserver(instance=another_instance)
+
+        invoice_month = self._generate_invoice_month()
+        expected_charges_details = {
+            'name': 'Mock AppServer charge',
+            'billing_start': invoice_month,
+            'billing_end': invoice_month,
+            'days': 20,
+            'charge': 12,
+        }
+        charges_details_mock.return_value = expected_charges_details
+        appservers_charges, appservers_total = get_instance_charges(desired_instance, invoice_month)
+
+        self.assertEqual(len(appservers_charges), 2)
+        for charge in appservers_charges:
+            self.assertEqual(charge, expected_charges_details)
+
+        self.assertEqual(appservers_total, 2 * expected_charges_details['charge'])
+
+    def test_generate_watched_forks_instances(self):
+        """
+        This will test that the method is able to return all of the watched
+        forks' instances for a specific organization and not anything else.
+        """
+        fork1, fork2 = self._setup_watched_forks()
+        watched_forks = generate_watched_forks_instances(self.organization)
+
+        self.assertIsInstance(watched_forks, defaultdict)
+        self.assertEqual(len(watched_forks), 2)
+
+        self.assertIsInstance(watched_forks[fork1], list)
+        self.assertEqual(len(watched_forks[fork1]), 2)
+        for instance in watched_forks[fork1]:
+            self.assertIsInstance(instance, OpenEdXInstance)
+
+        self.assertIsInstance(watched_forks[fork2], list)
+        self.assertEqual(len(watched_forks[fork2]), 1)
+        for instance in watched_forks[fork1]:
+            self.assertIsInstance(instance, OpenEdXInstance)
+
+    @mock.patch('reports.helpers.get_instance_charges')
+    def test_generate_charges(self, instance_charges_mock):
+        """
+        This method is to make sure the we're able to generate the charges
```

Comment:
- Simplify the wording (you can remove „This method is to…“)
- the→that.
- the description doesn't describe much more than the name. The description could explain what the test actually does, or could put the focus in some particular thing that is being tested (e.g. I think the point here is that it includes _all_ servers?).